### PR TITLE
Submit recipe for publish

### PR DESCRIPTION
**Pull request by CamronOlsen**

### CODE REVIEW

File: server/routes/private.js [hunk 1]
```diff
@@ -608,6 +609,46 @@ privateRouter.post('/users/:id/recipe/update_recipe', upload.single('userRecipeI
   }
 });
 
+privateRouter.post('/users/:id/recipe/request_publish', async (req, res) => {
+  const userId = req.params.id;
+  const user = await mongoose.model('User').findById(userId);
```
Comment (Thompson632) [inline server/routes/private.js:614]: Use `User.findById` here

File: server/routes/private.js [hunk 2]
```diff
@@ -608,6 +609,46 @@ privateRouter.post('/users/:id/recipe/update_recipe', upload.single('userRecipeI
...
+    }
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ error: "Error occurred sending publish request message." });
+  }
+});
+
+privateRouter.get('/users/:id/recipe/get_requests', async (_, res) => {
+  try {
+    const messages = await mongoose.model("RecipePubRequest").find({});
```
Comment (Thompson632) [inline server/routes/private.js:644]: Use `RecipePubRequest` here

File: client/public/static/js/recipes/recipe_details.js [hunk 3]
```diff
@@ -370,15 +370,21 @@ function RecipeDetailsView() {
...
-    button.addEventListener("click", function() {
+    updateRecipeButton.textContent = "Update Recipe";
+    updateRecipeButton.classList.add("updateRecipeButton"); // Add a class for styling
+    updateRecipeButton.addEventListener("click", function() {
       document.location.href = "/account/update_recipe?userRecipeName="+recipeName;
     });
     var container = document.getElementById("updateRecipeContainer");
-    container.appendChild(button);
+    container.appendChild(updateRecipeButton);
+
```
Comment (Thompson632) [inline client/public/static/js/recipes/recipe_details.js:382]: Could you please add a screenshot of what this looks like on recipe details? 

Also, what happens to the button when we submit? Can we submit multiple times? If so, we should disable the button if it's been "submitted"

**Comment by CamronOlsen** [comment]
Closing this out as all changes were pushed into recipe update pull req

**Pull request closed without merging**